pkg/kubelet/kubelet_pods_test.go

```diff
@@ -2358,3 +2359,180 @@ func TestTruncatePodHostname(t *testing.T) {
+					{
+						ContainerID: "test://1",
+						Name:        containerName,
+						State: v1.ContainerState{
+							Waiting: &v1.ContainerStateWaiting{
+								Reason:  "",
+								Message: "",
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "init container",
+			podSpec: v1.PodSpec{
+				Containers:     specContainerList,
+				InitContainers: specInitContainerList,
+			},
+			cStatuses: []*kubecontainer.ContainerStatus{
+				{
+					ID:   kubecontainer.BuildContainerID("test", "1"),
+					Name: containerName,
+				},
+				{
+					ID:   kubecontainer.BuildContainerID("test", "2"),
+					Name: initContainerName,
+				},
+			},
+			wantContainer: &v1.PodStatus{
+				PodIPs:   []v1.PodIP{},
+				QOSClass: v1.PodQOSBestEffort,
+				ContainerStatuses: []v1.ContainerStatus{
+					{
+						ContainerID: "test://1",
+						Name:        containerName,
+						State: v1.ContainerState{
+							Waiting: &v1.ContainerStateWaiting{
+								Reason:  "",
+								Message: "",
+							},
+						},
+					},
+				},
+				InitContainerStatuses: []v1.ContainerStatus{
+					{
+						ContainerID: "test://2",
+						Name:        initContainerName,
+						State: v1.ContainerState{
+							Waiting: &v1.ContainerStateWaiting{
+								Reason:  "",
+								Message: "",
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "ephemeral container",
+			podSpec: v1.PodSpec{
+				Containers:          specContainerList,
+				InitContainers:      specInitContainerList,
+				EphemeralContainers: ephemeralContainerList,
+			},
+			cStatuses: []*kubecontainer.ContainerStatus{
+				{
+					ID:   kubecontainer.BuildContainerID("test", "1"),
+					Name: containerName,
+				},
+				{
+					ID:   kubecontainer.BuildContainerID("test", "2"),
+					Name: initContainerName,
+				},
+				{
+					ID:   kubecontainer.BuildContainerID("test", "3"),
+					Name: ephemeralContainerName,
+				},
+			},
+			wantContainer: &v1.PodStatus{
+				PodIPs:   []v1.PodIP{},
+				QOSClass: v1.PodQOSBestEffort,
+				ContainerStatuses: []v1.ContainerStatus{
+					{
+						ContainerID: "test://1",
+						Name:        containerName,
+						State: v1.ContainerState{
+							Waiting: &v1.ContainerStateWaiting{
+								Reason:  "",
+								Message: "",
+							},
+						},
+					},
+				},
+				InitContainerStatuses: []v1.ContainerStatus{
+					{
+						ContainerID: "test://2",
+						Name:        initContainerName,
+						State: v1.ContainerState{
+							Waiting: &v1.ContainerStateWaiting{
+								Reason:  "",
+								Message: "",
+							},
+						},
+					},
+				},
+				EphemeralContainerStatuses: []v1.ContainerStatus{
+					{
+						ContainerID: "test://3",
+						Name:        ephemeralContainerName,
+						State: v1.ContainerState{
+							Waiting: &v1.ContainerStateWaiting{
+								Reason:  "",
+								Message: "",
+							},
+						},
+					},
+				},
+			},
+		},
+	} {
+		pod.Spec = test.podSpec
+		podStatus.ContainerStatuses = test.cStatuses
```

Comment:
Create a new `pod` and `podStatus` for each test case rather than reusing them here.